pkg/internal/resources/apiserver/apiserver.go

```diff
@@ -125,100 +124,105 @@ func Manifests(c Config) ([]unstructured.Unstructured, error) {
 			},
 		},
 	}
-	if c.Spec.StaticUsers != nil {
-		containers := deploymentPatch.Spec.Template.Spec.Containers
-		for i, container := range containers {
-			if container.Name == "manager" {
-				containers[i].Args = append(containers[i].Args,
-					"--htpasswd-secret-name=$(HTPASSWD_SECRET_NAME)",
-				)
-				containers[i].Env = append(containers[i].Env,
-					corev1.EnvVar{
-						Name:  "HTPASSWD_SECRET_NAME",
-						Value: c.Spec.StaticUsers.HtpasswdSecret.Name,
-					},
-				)
-				supportedAuth = append(supportedAuth, "Htpasswd")
+	var supportedAuth []string
+	for _, auth := range c.Spec.Authentication {
+		if auth.StaticUsers != nil {
+			containers := deploymentPatch.Spec.Template.Spec.Containers
+			for i, container := range containers {
```

Comment:
wait...you have only a single container, you're explicitly controlling it since it's a local variable, and you're looping over it finding the manager container? Could you make this simpler by just doing something like:

```container := &deploymentPatch.Spec.Template.Spec.Containers[0]``` in code somewhere. (You want to get the pointer to it, for all changes be reflected to that container)

Reply:
>wait...you have only a single container, you're explicitly controlling it since it's a local variable, and you're looping over it finding the manager container? Could you make this simpler by just doing something like

I think it's your code, I just leave it as is :)
but yeah, I can simplify it

Reply:
> I think it's your code

What daemon possessed me? :P